src/System.Private.ServiceModel/tools/scripts/StartWCFSelfHostedSvcDoWork.cmd

```diff
@@ -4,9 +4,12 @@ setlocal
 REM This script should always be called from StartWCFSelfHostedSvc.cmd
 
 set _setuplog=%~dp0..\..\..\..\SelfHostedWcfServiceSetup.log
+set _setupMonitorFile=%~dp0..\..\..\..\SelfHostedWcfServiceMonitor.log
 
 echo Preparing to launch the WCF self hosted service
 
+if EXIST %_setupMonitorFile% del %_setupMonitorFile%
```

Comment:
NIT NIT: naming... "monitor" is a little confusing when I first saw it. This is really a "semaphore" file.